tests/integration-service/namespace-backed-environments.go

```diff
@@ -240,10 +241,17 @@ var _ = framework.IntegrationServiceSuiteDescribe("Namespace-backed Environment
 
 		When("Integration Test PipelineRun completes successfully", func() {
 			It("should lead to Snapshot CR being marked as passed", FlakeAttempts(3), func() {
-				Eventually(func() bool {
+				Eventually(func() error {
 					snapshot, err = f.AsKubeAdmin.IntegrationController.GetSnapshot("", pipelineRun.Name, "", testNamespace)
```

Comment:
Thank you for this PR, this makes it easier to identify what went wrong, by improving the error messages. But I think the root cause is still yet to be fixed.

The main question is: Why does the `snapshot` variable contain a `nil` value?

I think it's because we're trying to fetch the Snapshot that's related to the older/outdated/wrong build PLR (stored within the `pipelineRun` variable). In all the occurrences of this bug, the `WaitForComponentPipelineToBeFinished()` function re-triggered (almost 3 times) Build PLR because the previous one(s) couldn't finish due to a timeout. Looking at the function signature of `WaitForComponentPipelineToBeFinished()`, it doesn't return them or make them available to the caller in any way. I think we should make changes to the `WaitForComponentPipelineToBeFinished` function to address this issue.

tagging @psturc to please cross-check if I'm making sense.

Reply:
I'm not sure if we need change the method signature - wouldn't it be sufficient to move [this line](https://github.com/redhat-appstudio/e2e-tests/pull/995/files#diff-41ef80bee886bb589e7d08af7a466e625c230cd50821a529ffcd5f8d284bf5b4R91) under the call of `WaitForComponentPipelineToBeFinished` method, which will guarantee that the `pipelineRun` variable value will be up to date?

Reply:
Are we sure that the first item of the `PipelineRunList` ([here](https://github.com/redhat-appstudio/e2e-tests/blob/ef0ffa41b4b6a41e60f02691783c2dfbea429bbc/pkg/clients/integration/pipelineruns.go#L82)) contains the latest PLR?

Reply:
Hello @dheerajodha and @psturc! Thank you so much for your insights!

I agree with both. I think we can move the `GetBuildPipelineRun` call under the `WaitForComponentPipelineToBeFinished` and ensure that we are getting the latest PLR. Going to address the changes